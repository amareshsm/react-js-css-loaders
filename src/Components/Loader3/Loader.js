import React from 'react'
import styles from './style.css'
const Loader3 = (props) => (
  <>
    <style>{`${styles}`}</style>
    <div className="loader-div three">
      <div className="loader3">
        <div className="loader-ring">
          <div></div>
          <div></div>
          <div></div>
          <div></div>
        </div>
      </div>
    </div>
  </>
)

export default Loader3
