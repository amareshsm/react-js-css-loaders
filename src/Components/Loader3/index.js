export { default as Loader3 } from './Loader'
