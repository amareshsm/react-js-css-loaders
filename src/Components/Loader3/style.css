.loader3 {
    box-sizing: border-box;
}

.loader3 {
    width: 80px;
    height: 80px;
}

.loader3 .loader-ring {
    display: inline-block;
    position: relative;
    width: 80px;
    height: 80px;
}

.loader3 .loader-ring div {
    display: block;
    position: absolute;
    width: 55px;
    height: 55px;
    border: 6px solid black;
    border-radius: 50%;
    border-color: black transparent transparent transparent;
    animation: spin 2s cubic-bezier(0.4, 0, 1, 1) infinite;
}
.loader3 .loader-ring div:nth-child(2) {
    animation-delay: -1s;
}

@keyframes spin {
    0% {
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(360deg);
    }
}
