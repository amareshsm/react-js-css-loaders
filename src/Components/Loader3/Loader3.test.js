import React from 'react'
import renderer from 'react-test-renderer'
import { Chaser } from '../Loader3'
it('renders correctly', () => {
  const tree = renderer.create(<Chaser />).toJSON()
  expect(tree).toMatchSnapshot()
})
