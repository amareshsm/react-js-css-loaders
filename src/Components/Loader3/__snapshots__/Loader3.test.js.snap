// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <style>
    [object Object]
  </style>,
  <div
    className="loader-div three"
  >
    <div
      className="loader3"
    >
      <div
        className="loader-ring"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>,
]
`;
